subprojects/core/src/main/java/org/gradle/tooling/provider/model/internal/DefaultIntermediateToolingModelProvider.java

```diff
@@ -38,47 +38,41 @@ public class DefaultIntermediateToolingModelProvider implements IntermediateTool
 
     private final IntermediateBuildActionRunner actionRunner;
     private final ToolingModelParameterCarrier.Factory parameterCarrierFactory;
+    private final ToolingModelProjectDependencyListener projectDependencyListener;
 
-    public DefaultIntermediateToolingModelProvider(IntermediateBuildActionRunner actionRunner, ToolingModelParameterCarrier.Factory parameterCarrierFactory) {
+    public DefaultIntermediateToolingModelProvider(
+        IntermediateBuildActionRunner actionRunner,
+        ToolingModelParameterCarrier.Factory parameterCarrierFactory,
+        ToolingModelProjectDependencyListener projectDependencyListener
+    ) {
         this.actionRunner = actionRunner;
         this.parameterCarrierFactory = parameterCarrierFactory;
+        this.projectDependencyListener = projectDependencyListener;
     }
 
     @Override
-    public <T> List<T> getModels(List<Project> targets, Class<T> modelType) {
-        return getModelsImpl(targets, modelType.getName(), modelType, null);
-    }
+    public <T> List<T> getModels(Project requester, List<Project> targets, String modelName, Class<T> modelType, Object modelBuilderParameter) {
```

Comment:
🤔 Might this `requester` name be replaced by more widely using `consumer`? Like you did in `ToolingModelProjectDependencyListener`